packages/nodes-base/nodes/Spotify/GenericFunctions.ts

```diff
@@ -0,0 +1,81 @@
+import { OptionsWithUri } from 'request';
+
+import {
+	IExecuteFunctions,
+	IHookFunctions,
+} from 'n8n-core';
+
+/**
+ * Make an API request to Spotify
+ *
+ * @param {IHookFunctions} this
+ * @param {string} method
+ * @param {string} url
+ * @param {object} body
+ * @param {object} query
+ * @returns {Promise<any>}
+ */
+export async function spotifyApiRequest(this: IHookFunctions | IExecuteFunctions, method: string, endpoint: string, body: object, query?: object): Promise<any> { // tslint:disable-line:no-any
+
+	let options: OptionsWithUri = {
+		method,
+		headers: {
+			'User-Agent': 'n8n',
+			'Content-Type': 'text/plain',
+			'Accept': ' application/json',
+		},
+		body,
+		qs: query,
+		uri: '',
+		json: true
+	};
+
+	try {
+		const authenticationMethod = this.getNodeParameter('authentication', 0, 'accessToken') as string;
+
+		const baseUrl = 'https://api.spotify.com/v1';
+
+		if (authenticationMethod === 'accessToken') {
```

Comment:
Nit, but you could potentially simplify by doing something like this:

```
const credentials = authenticationMethod === 'accessToken' ? this.getCredentials('spotifyApi') : credentials = this.getCredentials('spotifyOAuth2Api');

if (credentials === undefined) { ... }
if (Object.keys(body).length === 0) { ... }
options...
return ...;
```

Reply:
I think there is an extra `credentials =` after the colon, so
`const credentials = authenticationMethod === 'accessToken' ? this.getCredentials('spotifyApi') : this.getCredentials('spotifyOAuth2Api');`